### ISSUE TITLE
Upgrade ESO to golang 1.18

### ISSUE THREAD

**Issue by gusfcarvalho** [external-secrets/external-secrets#1064]
We should upgrade golang version from 1.17 to 1.18.

Not sure whether all of our dependencies already support it though, but we should track it down eventually.

Dependencies that we would need to update before going to 1.18:
* golangci-lint to [1.45](https://github.com/golangci/golangci-lint/issues/2649) at least
* ...(any others that we find)

**Comment by radiosilence** [comment]
If this can be done reasonably quickly, should it be high priority due to the fact that future code written might want to use 1.18 features (generics)? So it makes sense to do this perhaps before starting other work.

**Comment by gusfcarvalho** [comment]
Hi @radiosilence ! Unfortunately we depend on a lot of other kubernetes libraries to update to go1.18. It is not something we can do it ourselves right now. This issue is more to keep track on the future work when our dependencies themselves are already compatible.

**Comment by IdanAdar** [comment]
go.mod is now pointing to Go 1.19 so I believe this can be safely closed now. Please re-open if I got the request wrong.
